components/CourseFilter.tsx

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+type CourseFilterProps = {
+  handleCategoryChange: (category: string) => void;
+};
+
+const CourseFilter: React.FC<CourseFilterProps> = ({ handleCategoryChange }) => {
+  const [selectedCategory, setSelectedCategory] = useState('All');
+
+  const handleSelectChange = (category: string) => {
+    setSelectedCategory(category);
+    handleCategoryChange(category);
+  };
+
+  return (
+    <div className="w-1/2 mb-20">
+      <select
+        className="w-full border-2 border-slate-400 p-2 rounded-full focus-within:border-cyan-600 transition-colors"
```

Comment:
This styling actually does not work in Safari (see this [stackoverflow post](https://stackoverflow.com/questions/68258111/select-html-tag-wont-visible-same-in-chrome-and-safari). The class `appearance-none` would need to be used, which does remove the dropdown arrow `v`. The background-colour also seems different but I think that is okay.

Also I would suggest adding a bit more padding between the text and the side of the select box.

```suggestion
        className="appearance-none w-full border-2 border-slate-400 px-4 py-2 rounded-full focus-within:border-cyan-600 transition-colors"
```n